### PR TITLE
Support for etcdkeys as endpoints

### LINKED ISSUE #530: Can't create APIs with all HTTP method level endpoints
**Steps to reproduce:**

1) Create an API with the following OAS file. (**It has HTTP method level endpoints only**)

```
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
x-mgw-basePath: /petstore/v1
x-mgw-request-interceptor: validateRequest
paths:
  /dogs:
    post:
      x-mgw-production-endpoints:
        urls:
          - http://www.mocky.io/v2/5cdc00d52d0000dc49f5a4e9
      x-mgw-response-interceptor: validateResponse
      summary: Create a pet
      operationId: createPets
      responses:
        '201':
          description: Null response
  /cats:
    get:
      x-mgw-production-endpoints:
        urls:
          - http://www.mocky.io/v2/5cdc00bc2d00008415f5a4e8  
      summary: Info for a specific pet
      operationId: showPetById
      responses:
        '200':
          description: Expected response to a valid request
```

2) Build the project.

It gives the following error.

```
Compiling source
    src:0.0.0
error: src:0.0.0::Swagger_Petstore_1_0_0.bal:108:34: undefined symbol 'get_e6e3e3e9_7d70_4d9d_88a7_3e1858443bd2_sand'
error: src:0.0.0::Swagger_Petstore_1_0_0.bal:108:34: invalid remote function invocation, expected an client object
error: src:0.0.0::Swagger_Petstore_1_0_0.bal:196:34: undefined symbol 'post_e4a00f91_66cf_48e7_90fb_ccc444512c7f_sand'
error: src:0.0.0::Swagger_Petstore_1_0_0.bal:196:34: invalid remote function invocation, expected an client object
error: src:0.0.0::Swagger_Petstore_1_0_0.bal:284:34: undefined symbol 'get_d9defaf1_9d8d_4757_9d24_12cd2847063d_sand'
error: src:0.0.0::Swagger_Petstore_1_0_0.bal:284:34: invalid remote function invocation, expected an client object
```

### PR DESCRIPTION
**Pull request by VirajSalaka**
## Purpose
Etcd keys can be provided as production/sandbox endpoints. 

>x-mgw-production-endpoints:
        urls:
        - etcd(<etcdKey_1>, <defaultUrl_1>)
        - etcd(<etcdKey_2>,<defaultUrl_2>)

To setup the environment, user needs to execute the command as follows,
> ./gateway -e etcdUrl=<etcdUrl> -e etcdusername=<username> -e etcdpassword=<password> -e etcdtimer=10000 <path_to_balx>

if the following scenarios occur, etcd key will not be authenticated and will be redirected to the default Url.

- etcd server is not available
- etcd key is not provided

Additional:
Fixes #530 , #531

### CODE REVIEW

File: tests/src/test/java/org/wso2/micro/gateway/tests/common/CLIExecutor.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public void generate(String label, String project, String security) throws Excep
                         .getPath()).getAbsolutePath();
         System.setProperty("user.dir", path.toString());
 
-        String[] initArgs = {"init", project};
+        String[] initArgs = {"init", project,"--etcd"};
```
Comment (Rajith90) [inline tests/src/test/java/org/wso2/micro/gateway/tests/common/CLIExecutor.java]: WE need to remov --etcd from the init command

Comment (VirajSalaka) [reply tests/src/test/java/org/wso2/micro/gateway/tests/common/CLIExecutor.java]: fixed with the following commit.
https://github.com/wso2/product-microgateway/pull/534/commits/7dbe875cf38a438b93b25af35edede7c112641aa

Comment (VirajSalaka) [reply tests/src/test/java/org/wso2/micro/gateway/tests/common/CLIExecutor.java]: Fixed with https://github.com/wso2/product-microgateway/pull/534/commits/6ded66c3dbd7dbd90e8e54c093bc11732f3efc15 .

File: components/micro-gateway-core/src/main/ballerina/gateway/utils/etcd_util.bal [hunk 2]
```diff
@@ -158,7 +160,13 @@ public function etcdLookup(string base10EncodedKey) returns string {
         var msg = response.getJsonPayload();
         if(msg is json) {
             printDebug(KEY_ETCD_UTIL, "etcd responded with a payload");
-            base64EncodedValue = <string>msg.kvs[0].value;      
+            json value = msg.kvs[0].value;
+            if (value == null) {
+                io:println("json value is null.");
```
[the remark this answers is not in the document]
Comment (VirajSalaka) [reply components/micro-gateway-core/src/main/ballerina/gateway/utils/etcd_util.bal]: fixed with https://github.com/wso2/product-microgateway/pull/534/commits/6ddb79b8429e401def1d8f983a439a89cbaf63b2.

[the remark this answers is not in the document]
Comment (VirajSalaka) [reply components/micro-gateway-core/src/main/ballerina/gateway/utils/etcd_util.bal]: fixed with https://github.com/wso2/product-microgateway/pull/534/commits/07dd89411c1eece1bce72eca920c6bb14664fbd2.

**Pull request merged**